generate_dubbed_video_mappings.py

```diff
@@ -24,21 +24,25 @@ def dubbed_video_data_from_api(lang_code):
     videos = k.get_videos()
     return {v["youtube_id"]: v["translated_youtube_id"] for v in videos if v["youtube_id"] != v["translated_youtube_id"]}
 
-# TODO (arceduardvincent): Remove the CACHE_FILEPATH and replace it to None when done mapping the source code.
+
 def download_ka_dubbed_video_csv(download_url=None, cache_filepath=None):
-    download_url = "https://docs.google.com/spreadsheets/d/1k5xh2UXV3EchRHnYzeP6-YGKrmba22vsltGSuU9bL88/export?format=csv&id=1k5xh2UXV3EchRHnYzeP6-YGKrmba22vsltGSuU9bL88&gid=0"
-
-    # response = requests.get(download_url)
-    # if response.history:
-    #     print "Request was redirected"
-    #     for resp in response.history:
-    #         print resp.status_code, resp.url
-    #         download_url = resp.url
-    #     print "Final destination:"
-    #     print response.status_code, response.url
-    # else:
-    #     print "Request was not redirected"
-    #
+    """
+    Function to do the heavy lifting in getting the dubbed videos map.
+    """
+    # Get the redirect url
+    if not download_url:
+        logging.info("Getting spreadsheet location from Khan Academy")
+        khan_url = "http://www.khanacademy.org/r/translationmapping"
+        try:
+            download_url = urllib.request.urlopen(khan_url).geturl()
+            if "docs.google.com" not in download_url:
+                logging.warn("Redirect location no longer in Google docs (%s)" % download_url)
```

Comment:
We need to implement the retying to download the csv. So that it would`t fail at building language packs.